packages/mason/test/src/render_test.dart

```diff
@@ -172,5 +172,127 @@ void main() {
+          equals(expected),
+        );
+      });
+
+      test('headerCase outputs correct string', () {
+        const greeting = 'hello world';
+        const input = 'Greeting: {{greeting|headerCase}}!';
+        const expected = 'Greeting: Hello-World!';
+        expect(
+          input.render(<String, dynamic>{'greeting': greeting}),
+          equals(expected),
+        );
+      });
+
+      test('lowerCase outputs correct string', () {
+        const greeting = 'Hello World';
+        const input = 'Greeting: {{greeting|lowerCase}}!';
+        const expected = 'Greeting: hello world!';
+        expect(
+          input.render(<String, dynamic>{'greeting': greeting}),
+          equals(expected),
+        );
+      });
+
+      test('pascalCase outputs correct string', () {
+        const greeting = 'Hello World';
+        const input = 'Greeting: {{greeting|pascalCase}}!';
+        const expected = 'Greeting: HelloWorld!';
+        expect(
+          input.render(<String, dynamic>{'greeting': greeting}),
+          equals(expected),
+        );
+      });
+
+      test('paramCase outputs correct string', () {
+        const greeting = 'hello world';
+        const input = 'Greeting: {{greeting|paramCase}}!';
+        const expected = 'Greeting: hello-world!';
+        expect(
+          input.render(<String, dynamic>{'greeting': greeting}),
+          equals(expected),
+        );
+      });
+
+      test('pathCase outputs correct string', () {
+        const greeting = 'hello world';
+        const input = 'Greeting: {{greeting|pathCase}}!';
+        const expected = 'Greeting: hello/world!';
+        expect(
+          input.render(<String, dynamic>{'greeting': greeting}),
+          equals(expected),
+        );
+      });
+
+      test('sentenceCase outputs correct string', () {
+        const greeting = 'hello world';
+        const input = 'Greeting: {{greeting|sentenceCase}}!';
+        const expected = 'Greeting: Hello world!';
+        expect(
+          input.render(<String, dynamic>{'greeting': greeting}),
+          equals(expected),
+        );
+      });
+
+      test('snakeCase outputs correct string', () {
+        const greeting = 'hello world';
+        const input = 'Greeting: {{greeting|snakeCase}}!';
+        const expected = 'Greeting: hello_world!';
+        expect(
+          input.render(<String, dynamic>{'greeting': greeting}),
+          equals(expected),
+        );
+      });
+
+      test('titleCase outputs correct string', () {
+        const greeting = 'hello world';
+        const input = 'Greeting: {{greeting|titleCase}}!';
+        const expected = 'Greeting: Hello World!';
+        expect(
+          input.render(<String, dynamic>{'greeting': greeting}),
+          equals(expected),
+        );
+      });
+
+      test('upperCase outputs correct string', () {
+        const greeting = 'hello world';
+        const input = 'Greeting: {{greeting|upperCase}}!';
```

Comment:
Thanks so much for adding tests! We should probably add a few more tests to ensure chaining works properly.

```dart
  test('can be chained', () {
        const greeting = 'hello world';
        const input = 'Greeting: {{greeting|upperCase|dotCase}}!';
        const expected = 'Greeting: HELLO.WORLD!';
        expect(
          input.render(<String, dynamic>{'greeting': greeting}),
          equals(expected),
        );
      });
```

Reply:
Chaining is going to be out of scope for what I can contribute, but the concept looks great

Reply:
I don't think this will resolve as expected no matter what.

```
      test('upperCase outputs correct string', () {
        const greeting = 'HELLO WORLD';
        const input = 'Greeting: {{greeting|dotCase}}!';
        const expected = 'Greeting: HELLO.WORLD!';
        expect(
          input.render(<String, dynamic>{'greeting': greeting}),
          equals(expected),
        );
      });
```

This resolves to `'Greeting: hello.world!'`

Reply:
The implementation for dotCase is

```dart

String get dotCase => _getSnakeCase(separator: '.');

String _getSnakeCase({String separator: '_'}) {
  List<String> words = this._words.map((word) => word.toLowerCase()).toList();

  return words.join(separator);
}
```

More here:
https://github.com/techniboogie-dart/recase/blob/f562e4af869620586d20e1afdb0a07fc467dc194/lib/recase.dart#L56

 https://github.com/techniboogie-dart/recase/blob/f562e4af869620586d20e1afdb0a07fc467dc194/lib/recase.dart#L103-L107

Reply:
Got it, no worries! I'll spend some more time on this over the next few days, do you mind if I push additional commits to your branch? Thanks again!

Reply:
Sure thing. That all said, this now supports chaining

Reply:
> Yeah I'm leaning toward tweaking to: `{{name.dotCase().lowerCase()}}`.
> 
> wdyt?

My only concern is the parsing complexity of dealing with `{{domain.feature.variable.dotCase().lowerCase()}}` or some other dot naming convention one might try to use for their values... but this might not be a concern